libsweep/src/sweep.cc

```diff
@@ -288,111 +235,72 @@ void sweep_device_accumulate_scans(sweep_device_s device) {
       received = 1;
     }
   }
+} catch (const std::exception& e) {
+  // worker thread is dead at this point
```

Comment:
Just to be clear, I think this can b addressed in a separate PR, as this isn't something newly introduced by this PR